distributed/tests/test_worker_client.py

```diff
@@ -315,3 +315,16 @@ def test_submit_different_names(s, a, b):
         assert fut > 0
     finally:
         yield c.close()
+
+
+@gen_cluster(client=True)
+async def test_task_unique_groups_scatter(c, s, a, b):
```

Comment:
Not sure why, but occasionally when I run I get a timeouterror:

> E           tornado.util.TimeoutError: Operation timed out after 20 seconds

Reply:
How occasionally? I'm not seeing this after 10 runs.

Reply:
I was seeing it after 3 runs on my machine

Reply:
I tried converting Corey's example into a test and now I see that timeouterror every time. I wonder if pytest is masking the CancelledError somehow.

```python
@gen_cluster(client=True)
async def test_task_unique_groups_scatter_tree_reduce(c, s, a, b):
    from toolz import first

    def tree_reduce(objs):
        while len(objs) > 1:
            new_objs = []
            n_objs = len(objs)
            for i in range(0, n_objs, 2):
                inputs = objs[i : i + 2]
                obj = c.submit(sum, inputs)
                new_objs.append(obj)
            wait(new_objs)
            objs = new_objs

        return first(objs)

    for n_parts in [1, 2, 5, 10, 15]:
        a = await c.scatter(range(n_parts))
        b = tree_reduce(a)

        assert sum(range(n_parts)) == await b

        del a
```

Reply:
I've been seeing this timeout exception as well and what's very strange is that I've even increased the timeout to 50s and I get the exception long before 50s has passed. This is specifically true in the case of https://github.com/dask/dask/issues/6037

Reply:
I'm getting it almost immediately. Within a second.